src/IJulia.jl

```diff
@@ -15,17 +15,19 @@ const capture_stderr = true
 using ZMQ
 using JSON
 
-using Nettle
-const hmacstate = HMACState[]
+using MbedTLS
+const hmacstate = Ref{MbedTLS.MD{true}}()
+
 function hmac(s1,s2,s3,s4)
-    if isempty(hmacstate)
+    if !isdefined(hmacstate, :x)
         return ""
     else
-        update!(hmacstate[1], s1)
-        update!(hmacstate[1], s2)
-        update!(hmacstate[1], s3)
-        update!(hmacstate[1], s4)
-        return hexdigest!(hmacstate[1])
+        MbedTLS.reset!(hmacstate[])
+        for s in (s1, s2, s3, s4)
+            write(hmacstate[], s)
+        end
+        # Take the digest (returned as a byte array) and convert it to hex string representation
+        return join([hex(_, 2) for _ in MbedTLS.finish!(hmacstate[])])
```

Comment:
maybe
```
digest = MbedTLS.finish!(hmacstate[])
hexdigest = Array(UInt8, length(digest)*2)
for i = 1:length(digest)
    b = digest[i]
    d = b >> 4
    hexdigest[2i-1] = UInt8('0')+d+39*(d>9)
    d = b & 0xf
    hexdigest[2i] = UInt8('0')+d+39*(d>9)
end
return Compat.ASCIIString(hexdigest)
```
to avoid all of the temporary allocations.